sanic/asgi.py

```diff
@@ -195,6 +204,8 @@ async def __call__(self) -> None:
         Handle the incoming request.
         """
         try:
+            self.stage = Stage.HANDLER
             await self.sanic_app.handle_request(self.request)
         except Exception as e:
             await self.sanic_app.handle_exception(self.request, e)
+            self.stage = Stage.FAILED
```

Comment:
This should probably not enter `Stage.FAILED`. Just having an exception is not a failure, but an incomplete/unfinished response is. If either the handler or `handle_exception` managed to send a complete response, the stream should go to `Stage.IDLE` and continue normally.

Reply:
I added a finally block and set stage to idle, do you think that works?

Reply:
Resolved below in another note.